eq-author-api/src/validation/index.js

```diff
@@ -12,58 +11,58 @@ require("ajv-keywords")(ajv, "select");
 
 const validate = ajv.addSchema(schemas.slice(1)).compile(schemas[0]);
 
-module.exports = questionnaire => {
-  validate(questionnaire);
+const formatErrorMessage = (error, questionnaire) => {
+  if (error.sectionId) {
+    delete error.dataPath;
+    delete error.schemaPath;
+    delete error.keyword;
 
-  if (!validate.errors) {
-    return [];
+    return error;
   }
 
-  const errorMessages = validate.errors.filter(
-    err => err.keyword === "errorMessage"
+  const { dataPath, message } = error;
+
+  const splitDataPath = dataPath.split("/");
+  const field =
+    message === "ERR_QCODE_REQUIRED"
+      ? "qCode"
+      : message === "ERR_SECONDARY_QCODE_REQUIRED"
+      ? "secondaryQCode"
+      : splitDataPath.pop();
```

Comment:
Hi Tom. I realise this block was here before, but I feel we could improve it's readability by swapping it to a switch statement. What do you think?

```suggestion      
      let field = ""
      
      switch(message) {
          case "ERR_QCODE_REQUIRED":
             field = "qCode";
             break;
         case "ERR_SECONDARY_QCODE_REQUIRED":
             field = "secondaryQCode";
             break;
         default:
             field = splitDataPath.pop();  
      }
```

Reply:
Sure, agreed :-)